app/controllers/system/admin/instance/components_controller.rb

```diff
@@ -7,19 +7,32 @@ def edit #:nodoc:
   end
 
   def update #:nodoc:
-    @settings.update(params.require(:settings_effective))
-    if current_tenant.save
+    validate_params
+    if @settings.update(settings_components_params) && current_tenant.save
       redirect_to admin_instance_components_path, success: t('.success')
     else
-      @settings.errors = current_tenant.errors
+      if @settings.errors.empty?
+        current_tenant.errors.full_messages.each do |msg|
```

Comment:
This is here so if saving fails at `current_tenant.save` for some strange reason, user will have some feedback as to what the problem is. I'm not expecting `current_tenant.save` to fail. 

Instead of rendering `edit`, would it make more sense for an error to be thrown if either `@settings.update(settings_components_params)` or `current_tenant.save` fails since we are not expecting any errors from them?